src/components/DateRange/index.js

```diff
@@ -24,16 +24,25 @@ class DateRange extends Component {
     if (!selectedRange || !onChange) return {};
 
     let { startDate, endDate } = selectedRange;
-    if (!endDate) endDate = new Date(startDate);
     let nextFocusRange;
     if (!isSingleValue) {
       startDate = value.startDate;
       endDate = value.endDate;
     } else if (focusedRange[1] === 0) {
       // startDate selection
       const dayOffset = differenceInCalendarDays(endDate, startDate);
+      const calculateEndDate = () => {
+        if (moveRangeOnFirstSelection) {
+          return addDays(value, dayOffset);
+        }
+        // allow continous range to stay as-is
+        if (!endDate) {
+          return endDate;
+        }
+        return !isBefore(value, endDate) ? value : endDate;
```

Comment:
Would this also work here?
```suggestion
        if (!endDate || isBefore(value, endDate)) {
          return endDate;
        }
        return value;
```